tests/Utils/CoerceValueTest.php

```diff
@@ -396,12 +396,29 @@ public function testReturnsErrorForNonObjectType(): void
         $this->expectGraphQLError($result, 'Expected type TestInputObject to be an object.');
     }
 
-    public function testReturnsNoErrorForStdClassInput(): void
+    /**
+     * @dataProvider coercibleObjects
+     */
+    public function testReturnsNoErrorForObjectInput(object $object): void
     {
-        $result = Value::coerceValue((object) ['foo' => 123], $this->testInputObject);
+        $result = Value::coerceValue($object, $this->testInputObject);
         $this->expectValue($result, ['foo' => 123]);
     }
 
+    /**
+     * @return iterable<int, array{object}>
+     */
+    public function coercibleObjects(): iterable
+    {
+        yield [(object) ['foo' => 123]];
+
+        yield [
+            new class {
```

Comment:
do we want to support casting classes to array as well? 🤔

Reply:
Maybe just keep supporting only stdClass. Casting other classes to array (or even iterating over it) is rather an abomination I'd say.